src/Agent.Worker/Release/RetryExecutor.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
+{
+    public class RetryExecutor
+    {
+        private const int DefaultMaximumRetryCount = 5;
+
+        private const int DefaultMillisecondsToSleepBetweenRetries = 1000;
+
+        public int MaximumRetryCount { get; set; }
+
+        public int MillisecondsToSleepBetweenRetries { get; set; }
+
+        public Func<Exception, bool> ShouldRetryAction { get; set; }
```

Comment:
The properties are set in the constructor. (The default values are set). However no external types sets this property. This retry executor is been used in RM Service and test where the properties are configurable but optional. I just moved the code as it was required. It should be fine to keep as it. Its not dead code. Its being used. The only additional change I have made on this file is adding async methods to it.